### PR TITLE
Nft packs/remove random oracle

### PR DESCRIPTION
**Pull request by MetaMykola**

### CODE REVIEW

File: nft-packs/program/src/utils.rs [hunk 1]
```diff
@@ -248,27 +248,24 @@ pub fn empty_account_balance(
...
-
-    if clock.slot.error_sub(slot)? > MAX_LAG_SLOTS {
-        return Err(NFTPacksError::RandomOracleOutOfDate.into());
-    }
-
-    // Hash random value from the oracle with current slot and proving process data and receive new random u16
+    // Hash slot, current timestamp and value from last slothash and proving process data and receive new random u16
     let mut hasher = DefaultHasher::new();
-    hasher.write(oracle_random_value.as_ref());
-    hasher.write(proving_process.try_to_vec()?.as_ref());
```
Comment (VadimGrozinok) [inline nft-packs/program/src/utils.rs:267]: It shouldn't be deleted because there can couple of this instructions in one slot

**Review by VadimGrozinok — CHANGES_REQUESTED** [review]

File: nft-packs/program/src/utils.rs (hunk 1 above)
Comment (MetaMykola) [reply nft-packs/program/src/utils.rs:267]: fixed

File: nft-packs/program/src/processor/request_card_to_redeem.rs [hunk 2]
```diff
@@ -166,8 +170,13 @@ pub fn request_card_for_redeem(
         return Err(NFTPacksError::UserRedeemedAllCards.into());
     }
 
-    let random_value =
-        get_random_oracle_value(randomness_oracle_account, &proving_process, &clock)?;
+    // get slot hash 
+    let data = recent_slothashes_info.data.borrow();
+    let most_recent = array_ref![data, 8, 8];
+    let index = u64::from_le_bytes(*most_recent);
```
Comment (VadimGrozinok) [inline nft-packs/program/src/processor/request_card_to_redeem.rs:176]: We shouldn't convert hash into u64
Use recent slot hash as it is

File: nft-packs/program/src/utils.rs [hunk 3]
```diff
@@ -248,27 +248,26 @@ pub fn empty_account_balance(
...
+    
+    // last slothash
+    hasher.write_u64(index);
```
Comment (VadimGrozinok) [inline nft-packs/program/src/utils.rs:261]: And based on previous comment here also will be some change(we will use write() method instead of write_u64())

**Review by VadimGrozinok — CHANGES_REQUESTED** [review]

**Comment by VadimGrozinok** [comment]
Not yet
Working on it
We already doing manual testing

**Pull request merged**